s32/drivers/s32k3/Mcl/src/Lcu_Ip.c

```diff
@@ -0,0 +1,2260 @@
+#endif
+
+/* Check if Lcu_Ip.c file and "Lcu_Ip_Multicore.h" file are of the same Software version */
+#if ((LCU_IP_SW_MAJOR_VERSION_C != LCU_IP_MULTICORE_SW_MAJOR_VERSION) || \
+     (LCU_IP_SW_MINOR_VERSION_C != LCU_IP_MULTICORE_SW_MINOR_VERSION) || \
+     (LCU_IP_SW_PATCH_VERSION_C != LCU_IP_MULTICORE_SW_PATCH_VERSION) \
+    )
+    #error "Software Version Numbers of Lcu_Ip.c and "Lcu_Ip_Multicore.h" are different"
+#endif
+#endif /* STD_ON == LCU_IP_MULTICORE_IS_AVAILABLE */
+
+#ifndef DISABLE_MCAL_INTERMODULE_ASR_CHECK
+/* Check if Lcu_Ip.c file and SchM_Mcl header file are of the same Autosar version */
+#if ((LCU_IP_AR_RELEASE_MAJOR_VERSION_C != SCHM_MCL_AR_RELEASE_MAJOR_VERSION) || \
+     (LCU_IP_AR_RELEASE_MINOR_VERSION_C != SCHM_MCL_AR_RELEASE_MINOR_VERSION))
+    #error "AutoSar Version Numbers of Lcu_Ip.c and SchM_Mcl.h are different"
+#endif
+#endif /* DISABLE_MCAL_INTERMODULE_ASR_CHECK */
+
+
+#if (STD_ON == LCU_IP_IS_AVAILABLE)
+/*==================================================================================================
+*                           LOCAL TYPEDEFS (STRUCTURES, UNIONS, ENUMS)
+==================================================================================================*/
+
+/*==================================================================================================
+*                                          LOCAL MACROS
+==================================================================================================*/
+
+/*==================================================================================================
+*                                         LOCAL CONSTANTS
+==================================================================================================*/
+
+/*==================================================================================================
+*                                         LOCAL VARIABLES
+==================================================================================================*/
+
+/*==================================================================================================
+*                                        GLOBAL CONSTANTS
+==================================================================================================*/
+
+/*==================================================================================================
+*                                        GLOBAL VARIABLES
+==================================================================================================*/
+#define MCL_START_SEC_VAR_CLEARED_UNSPECIFIED_NO_CACHEABLE
+#include "Mcl_MemMap.h"
+
+/* Pointer to the LCU Initialization Configuration. Based on this configuration pointer,
+   the LCU Driver obtains all information for the Logic Cell and corresponding
+   Cell Configurations.
+   The Pointer is loaded when Lcu_Ip_Init() is called. */
+static const Lcu_Ip_InitType * Lcu_Ip_pxInit;
```

Comment:
all variables that are no-cacheable (ie. between `#define *START*NO_CACHEABLE` and `#define *STOP*NO_CACHEABLE`)  must be patched with `VAR_SEC_NOCACHE()` macro. Please do all patches in a separate commit after adding the driver sources. See https://github.com/zephyrproject-rtos/hal_nxp/commit/5c2af05d593698da333e56c564828814823dd081

Reply:
@manuargue I have discussed with Lcu and Trgmux IP owners and have resolved it.. kindly check !!